arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -405,8 +355,36 @@ int arm_core_mpu_get_max_domain_partition_regions(void)
 		_get_region_index_by_type(THREAD_DOMAIN_PARTITION_REGION);
 }
 
+/**
+ * This internal function checks if the region is user accessible or not.
+ *
+ * Note:
+ *   The caller must provide a valid region number.
+ */
+static inline int _is_user_accessible_region(u32_t r_index, int write)
```

Comment:
I know that this change was just due to the relocation of the function, however, it give me the opportunity to bring up a question: `_is_user_accessible_region(..)` appears to mix up design with run-time hardware status, explained inline in following comments. So I wonder whether we could have a unified behavior, instead.

Basically, this boils down to the following question: Since the type of region determines the permissions, why do we need to ask the hardware in runtime?

Reply:
can this change be done in another PR

Reply:
Yes